lighthouse-core/audits/resource-budget.js

```diff
@@ -0,0 +1,182 @@
+        requestCount: summary[type].count,
+        size: summary[type].size,
+      };
+    });
+    // Force third-party to be last, descending by size otherwise
+    const thirdPartyRow = rows.find(r => r.resourceType === 'third-party') || [];
+    const otherRows = rows.filter(r => r.resourceType !== 'third-party')
+      .sort((a, b) => {
+        return b.size - a.size;
+      });
+    return otherRows.concat(thirdPartyRow);
+  }
+
+  /**
+   * @param {LH.Budget} budget
+   * @param {Record<LH.Budget.ResourceType,ResourceEntry>} summary
+   * @return {Array<{resourceType: LH.Budget.ResourceType, label: string, requestCount: number, size: number, sizeOverBudget: number | undefined, countOverBudget: string | undefined}>}
+   */
+  static budgetTable(budget, summary) {
+    const resourceTypes = /** @type {Array<LH.Budget.ResourceType>} */ (Object.keys(summary));
+    return resourceTypes.map((type) => {
+      const resourceType = type;
+      const label = this.rowLabel(type);
+      const requestCount = summary[type].count;
+      const size = summary[type].size;
+
+      let sizeOverBudget;
+      let countOverBudget;
+
+      if (budget.resourceSizes) {
+        const sizeBudget = budget.resourceSizes.find(b => b.resourceType === type);
+        if (sizeBudget && (size > (sizeBudget.budget * 1024))) {
+          sizeOverBudget = size - (sizeBudget.budget * 1024);
+        }
+      }
+      if (budget.resourceCounts) {
+        const countBudget = budget.resourceCounts.find(b => b.resourceType === type);
+        if (countBudget && (requestCount > countBudget.budget)) {
+          const requestDifference = requestCount - countBudget.budget;
+          countOverBudget = str_(UIStrings.requestCount, {count: requestDifference});
+        }
+      }
+      return {
+        resourceType,
+        label,
+        requestCount,
+        size,
+        countOverBudget,
+        sizeOverBudget,
+      };
+    }).filter((row) => {
+      // Only resources with budgets should be included in the table
+      if (budget.resourceSizes) {
+        if (budget.resourceSizes.find(b => b.resourceType === row.resourceType)) return true;
+      }
+      if (budget.resourceCounts) {
+        if (budget.resourceCounts.find(b => b.resourceType === row.resourceType)) return true;
+      }
+      return false;
+    }).sort((a, b) => {
+      return (b.sizeOverBudget || 0) - (a.sizeOverBudget || 0);
+    });
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+
+    /** @type {Record<LH.Budget.ResourceType,{count:number, size:number}>} */
+    const summary = await ResourceSummary.request({devtoolsLog, URL: artifacts.URL}, context);
+
+    /** @type {LH.Budget | undefined} */
+    const budget = context.settings.budgets ? context.settings.budgets[0] : undefined;
+
```

Comment:
here i think we should bail if there was no budget. 

```js
    // Don't return results if no budget was provided or matched
    if (!budget) {
      return {
        score: 0,
        notApplicable: true
      }
    }
```

and your current renderer code works perfectly with this. :)